Dockerfile.alpine

```diff
@@ -9,40 +9,45 @@ RUN apk add --update \
       openrc \
       git \ 
       make \
+      su-exec \
     && rm -rf /var/cache/apk/* \
     \
-    && addgroup holochain -g 868 \
-    && adduser -g 'Standard holochain user.' -u 868 -G holochain -h /home/holochain -s /bin/sh -D holochain \
-    && mkdir -p ~holochain/.ssh \
-    && chmod 700 ~holochain/.ssh \
-    && mv /etc/profile.d/color_prompt /etc/profile.d/color_prompt.sh \
-    && sed -i -e 's#/bin/ash#/bin/sh#' /etc/passwd \
-    && mkdir -p /work/bin /work/golang \
-    \
-    && touch /etc/holochain.env_vars \
-    && echo 'export GOPATH=/work/golang' >> /etc/holochain.env_vars \
-    && echo 'export GOBIN=/home/holochain/bin' >> /etc/holochain.env_vars \
-    && echo 'export PATH=$GOPATH/bin:/usr/local/go/bin:$GOBIN:$PATH' >> /etc/holochain.env_vars \
-    && cat /etc/holochain.env_vars \
-    && chown -R holochain /work \
-    && su holochain -c 'source /etc/holochain.env_vars; \
-      go get -v -u github.com/whyrusleeping/gx \
-      && go get -v -d github.com/metacurrency/holochain \
-      && cd /work/golang/src/github.com/metacurrency/holochain \
-      && make deps'
+    && addgroup holochain -g 868 
+
+# Install gx and holochain and all their dependencies and hold them in a docker image
+## Using Docker env tools, because we are installing everything as root
+  ENV GOPATH=/work/golang \
+      GOBIN=/home/holochain/bin
+  ENV PATH=$GOPATH/bin:/usr/local/go/bin:$GOBIN:$PATH
+
+  RUN go get -v -u github.com/whyrusleeping/gx
+  RUN go get -v -d github.com/metacurrency/holochain
+
+  WORKDIR /work/golang/src/github.com/metacurrency/holochain
+  RUN make deps
 
 WORKDIR /work/golang/src/github.com/metacurrency/holochain
 
-# above should get cached, does not depend on source
-ADD . /work/holochain 
+# make our development files available to hc, on top of the cached image "so far" from above
+  ADD . /work/holochain 
+
+  # this rm line works well because it stops weird artifacts coming over from github.com/metacurrency/holochain::master
+  RUN rm -rf /work/golang/src/github.com/metacurrency/holochain/* \
+      && cp -pr /work/holochain/* /work/golang/src/github.com/metacurrency/holochain/ \
+      && chown -R root:holochain /work \
+      && chmod 775 -R /work 
+
+# compile and test our development files
+  RUN make; make bs
```

Comment:
Im looking at it from the perspective of a developer or a user who dont want something built if it doesnt makes sense for it to be built. assuming, from their perspective, the docker build isnt solely from the point of view of using hc